src/server/register.rs

```diff
@@ -91,26 +92,25 @@ async fn register(
 /// Returns an estimation of an optimal `gas_premium` and `gas_fee_cap`
 /// for a transaction considering the average premium, base_fee and reward percentile from
 /// past blocks
-/// This is an adaptation of ethers' `eip1559_default_estimator`:
-/// https://github.com/gakonst/ethers-rs/blob/5dcd3b7e754174448f9a8cbfc0523896609629f9/ethers-core/src/utils/mod.rs#L476
-async fn premium_estimation(signer: Arc<DefaultSignerMiddleware>) -> anyhow::Result<(U256, U256)> {
-    let base_fee_per_gas = signer
-        .get_block(ethers::types::BlockNumber::Latest)
+async fn premium_estimation(provider: Arc<Provider>) -> anyhow::Result<(u128, u128)> {
```

Comment:
i wonder, does alloy have better built in gas estimation? i can't imagine everyone that uses alloy does this

Reply:
Will take a quick search around.

Reply:
@sanderpick see the latest commit here where I switched to Alloy's gas estimation. Tested it locally with no problems.